src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,127 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_CAPACITY = 16;
+    private int capacity;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        capacity = DEFAULT_CAPACITY;
+        size = 0;
+        table = new Node[capacity];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= capacity * LOAD_FACTOR) {
+            resize();
+        }
+        boolean counter = true;
+        int hash = hash(key);
+        int index = hash % capacity;
+        if (table[index] == null) {
+            table[index] = new Node<>(key, value, null, hash);
+            size++;
+        } else {
+            Node<K, V> checkKey = table[index];
+            while (checkKey != null) {
+                if (hash == checkKey.hash) {
+                    if (key == null && key == checkKey.key) {
+                        checkKey.value = value;
+                        counter = false;
+                        break;
+                    } else if (key.equals(checkKey.key)) {
+                        checkKey.value = value;
+                        counter = false;
+                        break;
+                    }
+                }
+                checkKey = checkKey.next;
+            }
+            if (counter) {
+                Node<K, V> newNode = new Node<>(key, value, table[index], hash);
+                table[index] = newNode;
+                size++;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int hash = hash(key);
+        int index = hash % capacity;
+        Node<K, V> result = table[index];
+        while (result != null) {
+            if (hash == result.hash) {
+                if (key == null && key == result.key) {
+                    return result.value;
+                } else if (key.equals(result.key)) {
+                    return result.value;
+                }
+            }
+            result = result.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : key.hashCode() >>> 2;
+    }
+
+    private void resize() {
+        if (capacity <= Integer.MAX_VALUE / 2) {
```

Comment:
@Korfat дивись, тут ну дуже багато "лесніц" із if-ів. Спробуй зробити так:
```suggestion
        if (capacity > Integer.MAX_VALUE / 2) {
        return;
        }
```